### PR TITLE
fix travis-ci scripts

### PR DESCRIPTION
**Pull request by XenoAmess**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,15 +14,19 @@
...
-  - openjdk8
-  - openjdk11
-  - openjdk12
-  - openjdk13
-  - openjdk14  
-  - openjdk-ea
-
+cache:
+  directories:
+    - $HOME/.m2
```
Comment (kinow) [inline .travis.yml:19]: Oh, that sounds smart! I'm using something similar with GH actions but for JS and Ruby (caching `yarn.lock` and `Gemfile.lock`). And this little trick has sped up significantly my builds. :rocket:

**Comment by XenoAmess** [comment]
> along with this we should add
> ` before_cache: rm -rfv $HOME/.m2/repository/org/apache/commons/cli`

Is it necessary?  and why？

**Comment by ameyjadiye** [comment]
There is one cache per branch and language version/ compiler version/ JDK version etc.. thus to make sure you clean the previously compiled .class files from cached .m2

**Comment by XenoAmess** [comment]
> There is one cache per branch and language version/ compiler version/ JDK version etc.. thus to make sure you clean the previously compiled .class files from cached .m2

I knew what it means but, does this project contains self circulate? or in other words, does its dependencies contains itself?
If not then I don't think we need to erase that cache because it will be overwritten after build IMO.

**Comment by ameyjadiye** [comment]
No, It does not contain itself. and yes it will overwrite  previous jar, removal is to make sure you will always have newest build. I see this a good practice.

**Comment by XenoAmess** [comment]
> No, It does not contain itself. and yes it will overwrite previous jar, removal is to make sure you will always have newest build. I see this a good practice.

OK, I see no cost in doing so.

**Comment by garydgregory** [comment]
Closing PR: Cleaned up differently.

**Pull request closed without merging**